### PR TITLE
backport: Merge bitcoin#22359

### PR DESCRIPTION
**Pull request by vijaydasmp**
bitcoin backports

### CODE REVIEW

File: src/test/miner_tests.cpp [hunk 1]
```diff
@@ -23,6 +23,7 @@
 #include <util/system.h>
 #include <util/time.h>
 #include <validation.h>
+#include <versionbits.h>
```
Comment (UdjinM6) [inline src/test/miner_tests.cpp:26]: 22630: too early for this, we don't have 22277 which introduced the issue

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/test/miner_tests.cpp (hunk 1 above)
Comment (vijaydasmp) [reply src/test/miner_tests.cpp]: dropped 22630

**Review by UdjinM6 — APPROVED** [review]
utACK

**Comment by vijaydasmp** [comment]
Hello @knst @PastaPastaPasta requesting review

**Pull request merged**